render/src/umi/mod.rs

```diff
@@ -131,101 +162,139 @@ impl Umi {
+                    data_lang: properties.get("data_lang").cloned(),
+                    attributes: Some(current_line.attributes).filter(|s| !s.is_empty()),
+                    implicit_closed: properties
+                        .get("implicit_closed")
+                        .ok_or(ParserError::MissingProperty((
+                            "implicit_closed".into(),
+                            current_line.position,
+                        )))?
+                        .parse()
+                        .unwrap_or_default(),
+                    tick_len: properties
+                        .get("tick_len")
+                        .ok_or(ParserError::MissingProperty((
+                            "tick_len".into(),
+                            current_line.position,
+                        )))?
+                        .parse()
+                        .unwrap_or_default(),
+                    start: Position::new(1, 1),
+                    end: Position::new(1, 1),
                 };
-                Block::VerbatimBlock(verbatim)
+                Ok(Block::VerbatimBlock(verbatim))
             }
             "BulletList" => {
                 let mut bullet_list = BulletList {
                     entries: vec![],
-                    start: serde_json::from_str(attributes.get("start").unwrap()).unwrap(),
-                    end: serde_json::from_str(attributes.get("end").unwrap()).unwrap(),
+                    start: Position::new(1, 1),
+                    end: Position::new(1, 1),
                 };
 
                 let bullet_list_depth = current_line.depth;
                 let mut current_line_index = line + 1;
-                current_line = self.fetch_next_line(current_line_index).unwrap();
+                current_line = self.fetch_next_line(current_line_index).unwrap_or_default();
 
                 while current_line.depth > bullet_list_depth {
                     if current_line.depth == bullet_list_depth + 1 {
                         // Append Element to Bullet List
                         let block = self.read_row(current_line_index);
                         let bullet_list_entry = match block {
-                            Block::BulletListEntry(block) => block,
-                            _ => panic!(),
+                            Ok(Block::BulletListEntry(block)) => block,
+                            _ => break,
                         };
                         bullet_list.entries.append(&mut vec![bullet_list_entry]);
-                    } else {
-                        break;
                     }
 
                     current_line_index += 1;
                     let fetched = self.fetch_next_line(current_line_index);
                     if fetched.is_none() {
                         break;
                     }
-                    current_line = fetched.unwrap();
+                    current_line = fetched.unwrap_or_default();
```

Comment:
```suggestion
let Some(fetched) = self.fetch_next_line(current_lin_index) else { break; };
current_line = fetched;
```

to replace it up to line 249.
gets rid of `unwrap`.